File: project-{topic}-plus/src/main/java/com/alx/dataviz/model/Chart.java
```java
package com.alx.dataviz.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Entity
@Table(name = "chart")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(callSuper = true)
@ToString
public class Chart extends BaseEntity {

    public enum ChartType {
        BAR, LINE, PIE, SCATTER, TABLE
    }

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ChartType type;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "data_source_id", nullable = false)
    private DataSource dataSource;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "dashboard_id", nullable = false)
    private Dashboard dashboard;

    @Column(columnDefinition = "TEXT")
    private String configuration; // JSON string for chart configuration (e.g., x-axis, y-axis, colors)
}
```

File: project-{topic}-pro/backend/src/database/entities/Chart.ts
```typescript
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, ManyToOne } from "typeorm";
import { Dashboard } from "./Dashboard";
import { DataSource } from "./DataSource";

export enum ChartType {
    BAR = "bar",
    LINE = "line",
    PIE = "pie",
    SCATTER = "scatter",
    TABLE = "table"
}

@Entity("charts")
export class Chart {
    @PrimaryGeneratedColumn("uuid")
    id!: string;

    @Column()
    name!: string;

    @Column({
        type: "enum",
        enum: ChartType,
        default: ChartType.BAR
    })
    type!: ChartType;

    @Column("jsonb") // Stores charting library specific configuration (e.g., axis, legends, colors)
    configuration!: Record<string, any>;

    @Column({ type: "text" }) // SQL query or similar data retrieval instruction
    query!: string;

    @Column("uuid")
    dashboardId!: string;

    @ManyToOne(() => Dashboard, dashboard => dashboard.charts, { onDelete: 'CASCADE' })
    dashboard!: Dashboard;

    @Column("uuid")
    dataSourceId!: string;

    @ManyToOne(() => DataSource, dataSource => dataSource.id)
    dataSource!: DataSource;

    @CreateDateColumn({ type: "timestamp", default: () => "CURRENT_TIMESTAMP" })
    createdAt!: Date;

    @UpdateDateColumn({ type: "timestamp", default: () => "CURRENT_TIMESTAMP", onUpdate: "CURRENT_TIMESTAMP" })
    updatedAt!: Date;
}
```